sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -67,13 +66,9 @@ object DateTimeUtils {
 
   def defaultTimeZone(): TimeZone = TimeZone.getDefault()
 
-  private val computedTimeZones = new ConcurrentHashMap[String, TimeZone]
-  private val computeTimeZone = new JFunction[String, TimeZone] {
-    override def apply(timeZoneId: String): TimeZone = TimeZone.getTimeZone(timeZoneId)
-  }
-
   def getTimeZone(timeZoneId: String): TimeZone = {
-    computedTimeZones.computeIfAbsent(timeZoneId, computeTimeZone)
+    val zoneId = ZoneId.of(timeZoneId, ZoneId.SHORT_IDS)
+    TimeZone.getTimeZone(zoneId)
```

Comment:
Doesn't this reintroduce the original bottleneck?
For reference this was introduced in https://github.com/apache/spark/pull/17933

Reply:
As I see the commit avoids synchronisation by the TimeZone class via using `ConcurrentHashMap`. This shouldn't introduce new bottleneck because we will still use a cache based `ConcurrentHashMap` inside of ZoneOffset.

Reply:
Oh I see, `TimeZone.getTimeZone(String)` is synchronized but this is now calling`TimeZone.getTimeZone(ZoneId)` which isn't.

Yes this seems OK. The new code path entails a little more parsing and overhead vs just looking up the cached value, but this isn't a 'hot' path to begin with. I think it's worth removing this extra level of caching for simplicity, yes.